sdcm/nemesis.py

```diff
@@ -1710,6 +1710,93 @@ def disrupt_network_start_stop_interface(self):  # pylint: disable=invalid-name
         finally:
             self.target_node.remoter.run("sudo /sbin/ifup eth1")
 
+    def streaming_err_and_rebuild(self, task='decommission'):
+        """
+        Stop streaming task in middle and rebuild the data on the node.
+        """
+
+        def streaming_task_thread(nodetool_task='decommission'):
+            """
+            Execute some nodetool command to start persistent streaming
+            task: decommission | rebuild | repair
+            """
+            try:
+                self.target_node.run_nodetool(nodetool_task)
+            except Exception:  # pylint: disable=broad-except
+                self.log.debug('%s is stopped' % nodetool_task)
+
+        streaming_thread = threading.Thread(target=streaming_task_thread, kwargs={'nodetool_task': task})
+        streaming_thread.start()
+
+        def is_streaming_started(nodetool_task='decommission'):
+            start_logs = {'rebuild': 'Rebuild starts',
+                          'decommission': 'DECOMMISSIONING: starts', 'repair': 'Repair start'}
+            result = self.target_node.remoter.run(
+                'ps aux|grep nodetool.*%s |grep -v grep' % nodetool_task, ignore_status=True)
+            self.log.debug(result.stdout)
+            if nodetool_task not in result.stdout:
+                return False
+            streaming_logs = self.target_node.search_database_log(start_logs[nodetool_task], start_from_beginning=False)
+            self.log.debug(streaming_logs)
+            return len(streaming_logs) > 0
+
+        wait.wait_for(func=is_streaming_started, timeout=200, step=1,
```

Comment:
We should trigger the interruption in different timings not only right after it started.

Reply:
??

Reply:
we can wait for random between 10s to 10m